packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx

```diff
@@ -55,21 +59,33 @@ export const TooltipPlugin: React.FC<TooltipPluginProps> = ({
   }, [focusedPointIdx, focusedSeriesIdx]);
 
   useEffect(() => {
-    const plotMouseLeave = () => {
+    const plotMouseLeave = (e: MouseEvent) => {
       setCoords(null);
+      setIsActive(false);
+      handleMouseEvent(e);
+    };
+
+    const plotMouseEnter = (e: MouseEvent) => {
+      setIsActive(true);
+      handleMouseEvent(e);
     };
 
     if (plotCtx && plotCtx.plot) {
       plotCtx.plot.over.addEventListener('mouseleave', plotMouseLeave);
+      plotCtx.plot.over.addEventListener('mouseenter', plotMouseEnter);
+      if (sync === DashboardCursorSync.Crosshair) {
+        plotCtx.plot.over.classList.add('u-crosshair');
+      }
     }
 
     return () => {
       setCoords(null);
       if (plotCtx && plotCtx.plot) {
         plotCtx.plot.over.removeEventListener('mouseleave', plotMouseLeave);
+        plotCtx.plot.over.removeEventListener('mouseenter', plotMouseEnter);
       }
     };
-  }, [plotCtx.plot?.root, setCoords]);
+  }, [plotCtx.plot?.root, setCoords, setIsActive]);
```

Comment:
is it a typical pattern for this array to contain the setter functions rather than the actual states?

functionally it makes no difference, but `coords, isActive` feels a bit better to me.

Reply:
Good point - I just looked into this some more.

I don't think we want to change it to `coords, isActive` since each time those change, they would trigger the useEffect.

[After researching this](https://github.com/facebook/react/issues/19125), it looks like it is a good practice to include functions that could change in the dependency array, but setState doesn't need to be.